src/wrappers.js

```diff
@@ -100,3 +100,220 @@ wrappers[".google.protobuf.Any"] = {
         return this.toObject(message, options);
     }
 };
+
+// recursive .fromObject implementation for google.protobuf.Value
+function googleProtobufValueFromObject(object, create) {
+    if (object === null) {
+        return create({
+            kind: "nullValue",
+            nullValue: 0
+        });
+    }
+    if (typeof object === "number") {
+        return create({
+            kind: "numberValue",
+            numberValue: object
+        });
+    }
+    if (typeof object === "string") {
+        return create({
+            kind: "stringValue",
+            stringValue: object
+        });
+    }
+    if (Array.isArray(object)) {
+        var array = object.map(function(element) { return googleProtobufValueFromObject(element, create); });
+        return create({
+            kind: "listValue",
+            listValue: {
+                values: array
+            }
+        });
+    }
+    if (typeof object === "object") {
+        var fields = {},
+            names = Object.keys(object),
+            i = 0;
+        for (; i < names.length; ++i) {
+            fields[names[i]] = googleProtobufValueFromObject(object[names[i]], create);
+        }
+        return create({
+            kind: "structValue",
+            structValue: {
+                fields: fields
+            }
+        });
+    }
+    return undefined;
+}
+
+// recursive .toObject implementation for google.protobuf.Value
+function googleProtobufValueToObject(message) {
+    if (message.kind === "nullValue") {
+        return null;
+    }
+    if (message.kind === "numberValue") {
+        return message.numberValue;
+    }
+    if (message.kind === "stringValue") {
+        return message.stringValue;
+    }
+    if (message.kind === "listValue") {
+        return message.listValue.values.map(googleProtobufValueToObject);
+    }
+    if (message.kind === "structValue") {
+        if (!message.structValue.fields) {
+            return {};
+        }
+        var names = Object.keys(message.structValue.fields),
+            i = 0,
+            struct = {};
+        for (; i < names.length; ++i) {
+            struct[names[i]] = googleProtobufValueToObject(message.structValue["fields"][names[i]]);
+        }
+        return struct;
+    }
+    return undefined;
+}
+
+// custom wrapper for google.protobuf.Value
+wrappers[".google.protobuf.Value"] = {
+    fromObject: function(object) {
+        // heuristic: if an object looks like a regular representation of google.protobuf.Value,
+        // with all those stringValues, etc., just accept it as is for compatibility.
+        if (typeof object === "object" && object) {
+            // something that has just one property called stringValue, listValue, etc.,
+            // and possibly a property called kind, is likely an object we don't want to touch
+            var names = Object.keys(object);
+            if (names.length === 1 && names.match(/^(?:null|number|string|list|struct)Value$/) ||
```

Comment:
I think there are cases where `names.match()` will be called on an Array object that doesn't have the `match()` method. For example, you might have an `obj` value that is `{ content: text }` that results in a `names` array with a length of 1 (but no `match()` method).

Maybe add an undefined check here?

```
if (names.length === 1 && names.match && names.match(...) ...
```